MAJOR changes to models. Minor change to settings import

JerroldJV opened the pull request:
Built out all the model classes.  Changed settings_secret to import at
end, so certain settings could be overriden (e.g. database password).

JerroldJV committed:
MAJOR changes to models. Minor change to settings import

Built out all the model classes.  Changed settings_secret to import at
end, so certain settings could be overriden (e.g. database password).

rsfriedman commented on clueless/models.py:
```diff
@@ -1,13 +1,163 @@
+    spaceNorth = models.OneToOneField('self', related_name='spaceSouth')
+    spaceEast = models.OneToOneField('self', related_name='spaceWest')
+    spaceCollector = models.ForeignKey(SpaceCollection)
+
```
What are spaceNorth and spaceEast?

JerroldJV replied:
A reference to the space immediately north of the space, and a reference to the space immediately east of the space.  This should make for easy lookups for move validation (lookup space to North and East, reverse lookup space to South and West).

rsfriedman commented on clueless/models.py:
```diff
@@ -1,13 +1,163 @@
+    """
+    card_id = models.AutoField(primary_key=True) #had to override, due to multiple inheritence conflicts later
+    name = models.CharField(max_length=30)
+
```
I feel like we need a value here as well so you might have like id=1, name=Room, value= Library

JerroldJV replied:
Class instances will be created via the admin screen or via a database setup python script.  Each instance of Card will get it's own auto-generated ID at that point.  These model definitions just state the attributes (and ultimately database columns) that each class has.

rsfriedman replied:
Hmm wouldn't a value be the attribute though? I feel like we need to be saving that somewhere even if not in here

rsfriedman commented on clueless/models.py:
```diff
@@ -1,13 +1,163 @@
+	"""
+    background = models.CharField(max_length=50)
+    position = models.IntegerField()
+
```
what is background?

JerroldJV replied:
... I've got no idea.  This was in the original models.py Keith committed.  Keith- care to comment?

rsfriedman replied:
@kchason

rsfriedman commented on clueless/models.py:
```diff
@@ -1,13 +1,163 @@
+    Parent game object
+    """
+    players = models.ManyToManyField(Player)
+    caseFile = models.ForeignKey(WhoWhatWhere)
```
I might be reading this wrong but manytomany for players does that imply a player might be playing multiple games? is that allowed? or should this be a one to many (1 game to many players) relationship?

JerroldJV replied:
... Good point.  It may be easier to make a one to many here, and make separate player instances if a particular User is playing multiple games.

rsfriedman commented on clueless/settings.py:
```diff
@@ -118,3 +117,5 @@
 
 STATIC_URL = '/static/'
+
+from .settings_secret import *
```
these extra symbols concern me o_0

JerroldJV replied:
What extra symbols?

rsfriedman replied:
when i go to the files changed tab I see this: 
![image](https://cloud.githubusercontent.com/assets/20866481/19929446/ccc8db7c-a0d9-11e6-9e70-c7499c03883c.png)

JerroldJV replied:
Odd, I see that on my phone but I swear I didn't see that last night on my desktop.  Let me see if I can clear that out

JerroldJV replied:
Really strange.  Tried to edit in windows and it doesn't even show up.  If I delete that line and recreate it in the web editor it doesn't even allow me to commit because it doesn't think there is a change...

JerroldJV replied:
OOhhhhhhh- hover over that symbol.  It means that there isn't a newline character at the end of the file.  I'll add it and commit.

rsfriedman replied:
That's so weird...I can pull the branch and test when I'm home to make sure it didn't break anything, but maybe it doesn't matter.

JerroldJV committed:
Minor updates to models

Change spaceNorth and SpaceEast to nullable (since an object could be on
a corner/side of the board).  Changed player from a ManyToMany with game
to a OneToMany (one game per player).

JerroldJV commented on clueless/settings.py:
```diff
@@ -118,3 +117,5 @@
 
 STATIC_URL = '/static/'
+
+from .settings_secret import *
```
Really strange.  Tried to edit in windows and it doesn't even show up.  If I delete that line and recreate it in the web editor it doesn't even allow me to commit because it doesn't think there is a change...

JerroldJV committed:
Update settings.py

Added newline character at end of file
